src/versionbits.cpp

```diff
@@ -105,6 +104,42 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
     return state;
 }
 
+// return the numerical statistics of blocks signalling the specified BIP9 condition in this current period
+BIP9Stats AbstractThresholdConditionChecker::GetStateStatisticsFor(const CBlockIndex* pindexPrev, const Consensus::Params& params) const
+{
+    BIP9Stats obj;
+    
+    int elapsedBlocks;
+    int nPeriod = Period(params);
+    int nThreshold = Threshold(params);
+    
+    const CBlockIndex* currentIndex = pindexPrev;
+    
+    obj.period = nPeriod;
+    obj.threshold = nThreshold;
+
+    // Find beginning of period
+    if (pindexPrev != NULL) {
+        pindexPrev = pindexPrev->GetAncestor(pindexPrev->nHeight - ((pindexPrev->nHeight + 1) % nPeriod));
+        elapsedBlocks = currentIndex->nHeight - pindexPrev->nHeight;
+        obj.elapsed = elapsedBlocks;
```

Comment:
Not sure about this. After making this one line, it seems the same to me. Still need to get `nHeight` from each. Simpler to me to see "Current - Start = Elapsed"